iree/turbine/kernel/wave/constraints.py

```diff
@@ -445,13 +445,28 @@ class TilingConstraint(DistributionConstraint):
     adds an index constraint to the K-th dimension of a tensor of the form
     BLOCK_K * i, where i is the induction variable associated with the
     loop around dimension K.
+
+    The constraint can also specify that we want to start with init_symbol
+    and then update the value to next_symbol at the end of the iteration.
+    When only init_symbol and next_symbol are specified, the count is
+    meaningless.
     """
 
     dim: IndexExpr
-    tile_size: IndexExpr
+    tile_size: Optional[IndexExpr] = None
     induction_var: Optional[IndexExpr] = None
     iters: Optional[IndexExpr] = None
     start: IndexExpr = Integer(0)
+    init_symbol: Optional[IndexSymbol] = None
+    next_symbol: Optional[IndexSymbol] = None
+    condition: Optional[Callable] = None
```

Comment:
We can add this for next PR, but leaving a comment as a TODO/documenation. Would be good to have the init_symbol, next_symbol, and condition to be directly inserted inside the kernel within the `tkw.iterate` function call. This will allow us to iterate over the same dimension but different conditions

Reply:
I have a new version now that should address all of these.